src/fastapi/templates/main_form.html

```diff
@@ -1,19 +1,34 @@
+<!DOCTYPE html>
 <html>
+    <head>
+        <meta charset="utf-8">
+        <link rel="stylesheet" href="static/css/style.css">
+        <link rel="icon" type="image/png" sizes="32x32" href="static/images/bottle.jpg">
+        <title>Sea Bottle Jackal</title>
+    </head>
     <body>
+        <script src="static/js/scripts.js"></script>
         <h1>Jackal Incorporated</h1>
         <form method="POST" enctype="multipart/form-data">
-            <input type="file" name="file"><br>
-            <input type="submit" value="Submit">
+            <div id="uploadButton">
+                <p><label id="imageUploadLabel" for="imageUpload"> Choose picture </label>
+                <input id="imageUpload" name="file" type="file" accept="image/*"></p>
+                <button id="submitPicture" type="submit">Jackalify picture!</button>
+            </div>
         </form>
         {% if picture %}
-            <img src="static/{{picture}}">
+            <div>
+                <progress id="progressBar" max="100" value="0"></progress> <br>
+                <img id="picture" src="static/working/{{picture}}" onload="checkProgress();checkGif()">
+            </div>
         {% endif %}
         {% if video %}
-            <!-- <video controls="controls">
-                <source src="static/{{video}}" type='video/mp4"'>
-                <a href="static/{{video}}">Download</a>.
-            </video> -->
-            <img src="static/{{video}}">
+            <div id="gif">
+                <p>
+                    <a class="downloadLink" download href="static/working/{{video}}">Download GIF</a>
+                    <a class="downloadLink" download href="static/working/{{photo}}">Download jackalified photo</a>
```

Comment:
Gif creates before the image. If you want to download picture right after the end of the process you will get json file and only several seconds later you will be able to download png.